product_filters/models/product.py

```diff
@@ -63,65 +63,77 @@ class ProductCategory(models.Model):
+        and the value is the quantity of products found in that price range.
+        @rtype: dict
+
+        """
         prod_obj = self.env['product.template']
         ranges_obj = self.env['product.price.ranges'].search([])
         count_dict = {}
-        prod_ids = []
-        if category:
-            prod_ids = prod_obj.search(
-                [('public_categ_ids', 'child_of', int(category)),
-                 ('website_published', '=', True)])
-        if prod_ids:
-            for prod in prod_ids:
-                for ran in ranges_obj:
-                    if ran.upper > prod.list_price > ran.lower:
-                        if ran.id in count_dict.keys():
-                            count_dict[ran.id] += 1
-                        else:
-                            count_dict[ran.id] = 1
-                    if ran.id not in count_dict.keys():
-                        count_dict[ran.id] = 0
-            to_jsonfy = [{'id': k, 'qty': count_dict[k]} for k in count_dict]
-            return to_jsonfy
+        prod_ids = category and prod_obj.search(
+            [('public_categ_ids', 'child_of', int(category)),
+             ('website_published', '=', True)]) or []
+        for prod in prod_ids:
+            for ran in ranges_obj:
+                count_dict.setdefault(ran.id, 0)
+                if ran.upper > prod.list_price > ran.lower:
+                    count_dict[ran.id] += 1
+        to_jsonfy = [{'id': k, 'qty': count_dict[k]} for k in count_dict]
+        return to_jsonfy
 
     @api.model
     def _get_async_values(self, category):
+        """Get quantity of products per attribute value based on a given
+        category.
+
+        @param category: The category id on which the quantities will be
+        searched for.
+        @param: int
+
+        @return: List of dictionaries where the key is the id of the range
+        and the value is the quantity of products found in that price range.
+        @rtype: dict
+
+        """
         prod_obj = self.env['product.template']
         count_dict = {}
-        prod_ids = []
-        if category:
-            prod_ids = prod_obj.search(
-                [('public_categ_ids', 'child_of', int(category)),
-                 ('website_published', '=', True)])
-        if prod_ids:
-            for prod in prod_ids:
-                for line in prod.attribute_line_ids:
-                    for value in line.value_ids:
-                        if value.id in count_dict.keys():
-                            count_dict[value.id] += 1
-                        else:
-                            count_dict[value.id] = 1
-                        if value.id not in count_dict.keys():
-                            count_dict[value.id] = 0
-            to_jsonfy = [{'id': k, 'qty': count_dict[k]} for k in count_dict]
-            return to_jsonfy
+        prod_ids = category and prod_obj.search(
+            [('public_categ_ids', 'child_of', int(category)),
```

Comment:
if the param is expecte integer, do not conver here, you should convert it in the call of the method because this method ask explicitly for an integer.